app/models/order.rb

```diff
@@ -1,29 +1,10 @@
 class Order < ActiveRecord::Base
   # ASSOCIATIONS --------------------------------------------------------
-
   has_many :order_items
-  # rich on rails before helpers
+  has_one :buyer
   before_create :set_order_status
   before_save :update_subtotal
 
-  # VALIDATIONS ----------------------------------------------------------
-
-  # validates :buyer_name, presence: true
-  # validates :buyer_email, {presence: true}
-  # validates :buyer_address, presence: true
-  # validates :buyer_zip, presence: true,
-  #                       numericality: { only_integer: true },
-  #                       length: { minimum: 5 }
-  # validates :buyer_state, presence: true,
-  #                         length: { is: 2 }
-  # validates :buyer_city, presence: true
-  # validates :buyer_expcc, presence: true
-  # validates :buyer_last4cc, presence: true,
-  #                           numericality: { only_integer: true },
-  #                           length: { is: 4 }
-
-  # validate :email_must_contain_at
-
   def subtotal
     order_items.collect { |oi| oi.valid? ? (oi.quantity * oi.unit_price) : 0 }.sum
```

Comment:
All of this logic should be separated out into several lines. Too many operations happening here.